rct229/rulesets/ashrae9012019/section5/section5rule1.py

```diff
@@ -0,0 +1,280 @@
+            rmd_p = context.PROPOSED
+            rmd_u = context.USER
+
+            has_proposed = True if find_one("$.type", rmd_p) else False
+            has_user = True if find_one("$.type", rmd_u, False) else False
+
+            baseline_rmd_list = [rmd_b0, rmd_b90, rmd_b180, rmd_b270]
+            rmd_list = [rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p, rmd_u]
+
+            # count the rmds that aren't None
+            no_of_rmds = len(
+                filter_(
+                    rmd_list,
+                    lambda rmd: rmd is not None,
+                )
+            )
+
+            # filter out baseline rmds that have the type key
+            baseline_list = [
+                find_one("$.type", rmd_b)
+                for rmd_b in baseline_rmd_list
+                if find_one("$.type", rmd_b, False)
+            ]
+
+            no_of_output_instance = len(baseline_list)
+
+            return {
+                "has_proposed": has_proposed,
+                "has_user": has_user,
+                "baseline_list": baseline_list,
+                "no_of_rmds": no_of_rmds,
+                "no_of_output_instance": no_of_output_instance,
+            }
+
+        class BuildingRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule1.RMDRule.BuildingRule, self).__init__(
+                    rmrs_used=produce_ruleset_model_instance(
+                        USER=True,
+                        BASELINE_0=True,
+                        BASELINE_90=True,
+                        BASELINE_180=True,
+                        BASELINE_270=True,
+                        PROPOSED=True,
+                    ),
+                    required_fields={
+                        "$.building_segments[*].zones[*].surfaces[*]": ["azimuth"],
+                    },
+                    fail_msg="Fail unless Table G3.1#5a exception #2 is applicable and it can be demonstrated that the building orientation is dictated by site considerations.",
+                )
+
+            def get_calc_vals(self, context, data=None):
+                building_b0 = context.BASELINE_0
+                building_b90 = context.BASELINE_90
+                building_b180 = context.BASELINE_180
+                building_b270 = context.BASELINE_270
+                building_p = context.PROPOSED
+                building_u = context.USER
+
+                has_proposed = data["has_proposed"]
+                has_user = data["has_user"]
+                baseline_list = data["baseline_list"]
+                no_of_rmds = data["no_of_rmds"]
+                no_of_output_instance = data["no_of_output_instance"]
+
+                has_baseline_0 = RULESET_MODEL.BASELINE_0 in baseline_list
+                has_baseline_90 = RULESET_MODEL.BASELINE_90 in baseline_list
+                has_baseline_180 = RULESET_MODEL.BASELINE_180 in baseline_list
+                has_baseline_270 = RULESET_MODEL.BASELINE_270 in baseline_list
```

Comment:
I kind of feel the baseline_list is not necessary because context.BASELINE_X makes sure the context is the correct type.
You can do this without even use the baseline_list by simply checking whether `building_bx` is None or an Object.
Similarly, you do not need `has_proposed` or `has_user` from higher level because ``building_p` and `building_u` are from these two models. If they are None, then it means there are no proposed or user.

Reply:
Addressed